src/change/models/changeManager.ts

```diff
@@ -0,0 +1,124 @@
+      }
+    }
+    case 'LineString': {
+      feature = feature as FlattenedGeoJSONLine;
+      switch (action) {
+        case Actions.CREATE: {
+          return getChangeFromLine({ action, feature });
+        }
+        case Actions.MODIFY: {
+          const oldElement = getOsmWayFromElements(osmElements);
+          return getChangeFromLine({ action, feature, oldElement });
+        }
+        case Actions.DELETE: {
+          const oldElement = getOsmWayFromElements(osmElements);
+          return getChangeFromLine({ action, oldElement });
+        }
+      }
+    }
+    case 'Polygon': {
+      feature = feature as FlattenedGeoJSONPolygon;
+      switch (action) {
+        case Actions.CREATE: {
+          return getChangeFromPolygon({ action, feature });
+        }
+        case Actions.MODIFY: {
+          const oldElement = getOsmWayFromElements(osmElements);
+          return getChangeFromPolygon({ action, feature, oldElement });
+        }
+        case Actions.DELETE: {
+          const oldElement = getOsmWayFromElements(osmElements);
+          return getChangeFromPolygon({ action, oldElement });
+        }
+      }
+    }
+  }
+};
+
+export const getNodeFromElements = (elements: OsmApiElements): OsmNode => {
+  if (!validateArrayHasElements(elements)) {
+    throwParseOsmElementsError('node');
+  }
+  const node = elements[0];
+  if (!isNode(node)) {
+    throwParseOsmElementsError('node');
+  }
+  return node as OsmNode;
+};
+
+export const getOsmWayFromElements = (elements: OsmApiElements): OsmWay => {
+  const osmWay = parseOsmWayApi(elements);
+  if (osmWay === undefined) {
+    throwParseOsmElementsError('way');
+  }
+  return osmWay as OsmWay;
+};
+
+/**
+ * Gets the tempOsmId generated in the change, used on Create action.
+ * @param {BaseElement[]} elements - Expected to be an array of either one OsmNode or one OsmWay and multiple OsmNodes
+ * @returns {number} The id of either the only existing OsmNode element or OsmWay element
+ */
+export const getTempOsmId = (elements: BaseElement[]): number => {
+  if (elements.length === 1) {
+    return elements[0].id;
+  }
+  const way = elements.find(isWay);
+  return (way as OsmWay).id;
+};
+
+export const throwParseOsmElementsError = (elementType: OsmElementType): never => {
+  throw new ParseOsmElementsError(`Could not parse osm-api-elements, expected ${elementType as string} element`);
+};
+
+@injectable()
+export class ChangeManager {
+  public constructor(@inject(Services.LOGGER) private readonly logger: ILogger) {}
+  public generateChange(action: Actions, geojson: FlattenedGeoJSON, osmElements: OsmApiElements, externalId: string): ChangeModel {
+    const osmChange: OsmChange = generateOsmChange(action, geojson, osmElements);
+    let changeModel: ChangeModel = {
+      action,
+      change: osmChange,
+      externalId,
+    };
+    if (action === Actions.CREATE && osmChange.create) {
+      const tempOsmId = getTempOsmId(osmChange.create);
+      changeModel = {
```

Comment:
a `changeModel` is being created here. `change` is one of the properties of `changeModel` and represents an `osmChange`.
should we rename those to avoid confusion?